src/wallet/wallet.cpp

```diff
@@ -245,7 +245,7 @@ bool CWallet::Unlock(const SecureString& strWalletPassphrase, bool anonymizeOnly
 {
     SecureString strWalletPassphraseFinal;
 
-    if (!IsLocked())
+    if (!IsLocked() && !fWalletUnlockAnonymizeOnly) // was already fully unlocked, not only for anonymization
```

Comment:
Looking more at that code, I think even more stuff should be changed. Thanks for review 👍
I'll dig a bit more.